bundle/Controller/LocationController.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\EzContentOnTheFlyBundle\Controller;
+
+use eZ\Publish\API\Repository\ContentService;
+use eZ\Publish\API\Repository\ContentTypeService;
+use eZ\Publish\API\Repository\Repository;
+use eZ\Publish\Core\Base\Exceptions\UnauthorizedException;
+use eZ\Publish\Core\Base\Exceptions\NotFoundException;
+use eZ\Publish\API\Repository\LocationService;
+use eZ\Publish\Core\REST\Server\Controller;
+use eZ\Publish\Core\REST\Server\Values;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+class LocationController extends Controller
+{
+    protected $locationService;
+
+    protected $contentTypeService;
+
+    protected $contentService;
+
+    protected $configuration;
+
+    protected $logger;
+
+    public function __construct(
+        Repository $repository,
+        LocationService $locationService,
+        ContentTypeService $contentTypeService,
+        ContentService $contentService,
+        LoggerInterface $logger)
+    {
+        $this->repository = $repository;
+        $this->locationService = $locationService;
+        $this->contentTypeService = $contentTypeService;
+        $this->contentService = $contentService;
+        $this->logger = $logger;
+    }
+
+    public function suggestedAction(Request $request, $content)
+    {
+        if (!isset($this->configuration[$content]) && $content != 'default') {
+            $content = 'default';
+        }
+
+        if (isset($this->configuration[$content])) {
+            $locations = $this->configuration[$content]['location'];
+        } else {
+            $locations = [];
+        }
+
+        $contentType = $this->contentTypeService->loadContentTypeByIdentifier($content);
+        $contentCreateStruct = $this->contentService->newContentCreateStruct($contentType, 'eng-GB');
```

Comment:
How about this way: https://github.com/ezsystems/content-on-the-fly-prototype-bundle/pull/5/commits/15a17671ca9fefa548cdc296232f21bdda6e9290 ?